src/tables/stat.rs

```diff
@@ -0,0 +1,336 @@
+            value_name_id: s.read::<u16>()?,
+            nominal_value: s.read::<Fixed>()?,
+            range_min_value: s.read::<Fixed>()?,
+            range_max_value: s.read::<Fixed>()?,
+        })
+    }
+}
+
+/// Axis value table format 3
+#[derive(Clone, Copy, Debug)]
+pub struct AxisValueTableFormat3 {
+    /// Zero-based index into [`Table::axes`].
+    pub axis_index: u16,
+    /// Flags for AxisValue.
+    pub flags: AxisValueFlags,
+    /// The name ID of the display string.
+    pub value_name_id: u16,
+    /// Numeric value for this record.
+    pub value: Fixed,
+    /// Numeric value for a style-linked mapping.
+    pub linked_value: Fixed,
+}
+
+impl FromData for AxisValueTableFormat3 {
+    const SIZE: usize = 14;
+
+    #[inline]
+    fn parse(data: &[u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+        Some(AxisValueTableFormat3 {
+            axis_index: s.read::<u16>()?,
+            flags: AxisValueFlags(s.read::<u16>()?),
+            value_name_id: s.read::<u16>()?,
+            value: s.read::<Fixed>()?,
+            linked_value: s.read::<Fixed>()?,
+        })
+    }
+}
+
+/// Axis value table format 4
+#[derive(Clone, Copy, Debug)]
+pub struct AxisValueTableFormat4<'a> {
+    /// Flags for AxisValue.
+    pub flags: u16,
+    /// The name ID of the display string.
+    pub value_name_id: u16,
+    /// List of axis-value pairings.
+    pub values: LazyArray16<'a, AxisValue>,
+}
+
+impl<'a> AxisValueTableFormat4<'a> {
+    fn parse(data: &'a [u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+        let axis_count = s.read::<u16>()?;
+        let flags = s.read::<u16>()?;
+        let value_name_id = s.read::<u16>()?;
+        let values = s.read_array16::<AxisValue>(axis_count)?;
+
+        Some(AxisValueTableFormat4 {
+            flags,
+            value_name_id,
+            values,
+        })
+    }
+}
+
+/// An [axis value table](https://learn.microsoft.com/en-us/typography/opentype/spec/stat#axis-value-tables).
+#[allow(missing_docs)]
+#[derive(Clone, Copy, Debug)]
+pub enum AxisValueTable<'a> {
+    Format1(AxisValueTableFormat1),
+    Format2(AxisValueTableFormat2),
+    Format3(AxisValueTableFormat3),
+    Format4(AxisValueTableFormat4<'a>),
+}
+
+impl FromData for AxisRecord {
+    const SIZE: usize = 8;
+
+    #[inline]
+    fn parse(data: &[u8]) -> Option<Self> {
+        let mut s = Stream::new(data);
+        Some(AxisRecord {
+            tag: s.read::<Tag>()?,
+            name_id: s.read::<u16>()?,
+            ordering: s.read::<u16>()?,
+        })
+    }
+}
+
+/// A [Style Attributes Table](https://docs.microsoft.com/en-us/typography/opentype/spec/stat).
+#[derive(Clone, Copy, Debug)]
+pub struct Table<'a> {
+    /// List of axes
+    pub axes: LazyArray16<'a, AxisRecord>,
+    /// Fallback name when everything can be elided.
+    pub fallback_name_id: Option<u16>,
+    /// Version of the style attributes table.
+    pub version: u32,
```

Comment:
Do we have to expose the version?

Reply:
It depends on how strict the parsing should be.  v1.2 introduced a new axis value table type (format 4).  If it's okay to accept format 4 entries in an older version of the `STAT` table then we can drop the version.

Reply:
Checking version during parsing is fine, I just don't see why this field should be public. Version should be handled implicitly by the parser.